### PR TITLE
Fix 2 edit workshop bugs discovered in React class refactor

### PR DESCRIPTION
**Pull request by aoby**
I discovered 2 long-standing bugs :( while testing React refactors [here](https://github.com/code-dot-org/code-dot-org/pull/18103#discussion_r142549176).

### Bug:
sessions-removal-bug

### Fix:
sessions-removal-fixed

### Bug:
last-facilitator-bug

### Fix:
last-facilitator-fixed

It also still works correctly when none are available:
image

Or when no course is selected:
image

### CODE REVIEW

File: apps/src/code-studio/pd/workshop_dashboard/components/facilitator_list_form_part.jsx [hunk 1]
```diff
@@ -60,14 +60,18 @@ export default class FacilitatorListFormPart extends React.Component {
       this.props.facilitators.push({id: -1});
     }
 
+    let filteredAvailableFacilitators = null;
+    // Remove already-selected facilitators from available list.
+    if (!this.props.readOnly) {
+      filteredAvailableFacilitators = this.props.availableFacilitators.filter(availableFacilitator => {
+        return !this.props.facilitators.find(f => f.id === availableFacilitator.id);
+      });
+    }
```
Comment (aoby) [inline apps/src/code-studio/pd/workshop_dashboard/components/facilitator_list_form_part.jsx]: Moving this out of the `map` on line 71 is just an optimization, not related to the bugs, but it seems like a good idea. There's really no need to recompute this available list for each row.

Comment (islemaster) [reply apps/src/code-studio/pd/workshop_dashboard/components/facilitator_list_form_part.jsx]: :+1: Good extraction.  One syntax nit - could express the outer arrow function here with an expression too.
```jsx
filteredAvailableFacilitators = this.props.availableFacilitators.filter(availableFacilitator =>
  !this.props.facilitators.find(f => f.id === availableFacilitator.id);
);
```

File: apps/src/code-studio/pd/workshop_dashboard/components/facilitator_list_form_part.jsx [hunk 2]
```diff
@@ -105,15 +110,19 @@ export default class FacilitatorListFormPart extends React.Component {
     }
 
     let addButton = null;
+    // Only show add button on the last row, when we're below the max,
+    // when it's not a placeholder (id -1), and we have at least one available to add
     if (i === facilitators.length - 1 && facilitators.length < MAX_FACILITATORS &&
-      this.props.facilitators[i].id > 0 && filteredAvailableFacilitators.length > 1 ) {
+      this.props.facilitators[i].id > 0 && filteredAvailableFacilitators.length >= 1 ) {
```
Comment (islemaster) [inline apps/src/code-studio/pd/workshop_dashboard/components/facilitator_list_form_part.jsx:116]: There might be another bug here - if the placeholder id is -1, shouldn't the third condition here be `this.props.facilitators[i] >= 0` not `> 0`?

File: apps/src/code-studio/pd/workshop_dashboard/components/facilitator_list_form_part.jsx (hunk 1 above)
Comment (aoby) [reply apps/src/code-studio/pd/workshop_dashboard/components/facilitator_list_form_part.jsx]: ah right, good call 👍  Will do

File: apps/src/code-studio/pd/workshop_dashboard/components/facilitator_list_form_part.jsx (hunk 2 above)
Comment (aoby) [reply apps/src/code-studio/pd/workshop_dashboard/components/facilitator_list_form_part.jsx:116]: Rails ids start at 1, so in this case either way will work. I'd be happy to change it to >= 0 if you think that's more readable.

Comment (islemaster) [reply apps/src/code-studio/pd/workshop_dashboard/components/facilitator_list_form_part.jsx:116]: TIL. LGTM.

Comment (aoby) [reply apps/src/code-studio/pd/workshop_dashboard/components/facilitator_list_form_part.jsx:116]: Or maybe it's due to the underlying [MySql auto increment](https://dev.mysql.com/doc/refman/5.7/en/example-auto-increment.html), but anyway this is the behavior - our ids always start with 1.

**Pull request merged**